File: python/src/selection_sort.py
from typing import List

Numbers = List[int]

def selection_sort(numbers: Numbers):
	for index, number in enumerate(numbers):
		comparison_numbers = numbers[index + 1:]

		minimum_number = {
			"value": number,
			"index": index
		}

		for sub_index, current in enumerate(comparison_numbers):
			sub_index += index + 1
			if current < minimum_number["value"]:
				minimum_number["value"], minimum_number["index"] = current, sub_index

		min_num = minimum_number["value"]
		min_num_index = minimum_number["index"]

		if number != minimum_number["value"]:
			numbers[min_num_index], number[index] = number, min_num

	return numbers

if __name__ == "__main__":
	print(
		selection_sort([29, 72, 98, 13, 87, 66, 52, 51, 36])
	)

File: typescript/src/selection-sort.ts
function selectionSort(numbers: Array<number>) {
	for (let i in numbers) {
		const index = Number(i)
		const comparisonNumbers = numbers.slice(index + 1)
		const currentNumber = numbers[index]

		const indexValue = index
		const minimumNumber = { value: currentNumber, indexValue}

		for (let j in comparisonNumbers) {
			const subIndex = Number(j) + indexValue + 1
			const current = comparisonNumbers[Number(j)]

			if (current < minimumNumber.value) {
				minimumNumber.value = current
				minimumNumber.indexValue = subIndex
			}
		}

		if (currentNumber != minimumNumber.value) {
			const greater = currentNumber
			numbers[minimumNumber.indexValue] = greater
			numbers[index] = minimumNumber.value
		}
	}

	return numbers
}

console.log(
	selectionSort([29, 72, 98, 13, 87, 66, 52, 51, 36])
)